### PR TITLE
feat: replacing old debugging project with new TODO app project

### PR DESCRIPTION
**Pull request by jdwilkin4**
I added some comments in the JS file for potential debugging steps that we can have campers do. 

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

## Live example
https://kvxp3h.csb.app/

## Concepts covered

- [x] getElementById
- [x] const and let
- [x] basic array and object usage
- [x] addEventListener
- [x] arrow functions
- [x]   e.preventDefault();
- [x] unshift
- [x] split
- [x] join
- [x] localStorage set item
- [x] localStorage get item
- [x] JSON.stringify
- [x] innerHTML
- [x] template strings
- [x] array.map
- [x] object destructuring 
- [x] remove()
- [x] findIndex
- [x] ===
- [x] splice
- [x] JSON.parse

### CODE REVIEW

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  localStorage.setItem("data", JSON.stringify(taskData));
+
+  addTaskToTaskContainer();
+};
+
+const addTaskToTaskContainer = () => {
+  tasksContainer.innerHTML = "";
+
+  taskData.map(
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js:55]: This should be a `.forEach` as we're not using the returned array.

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  let dataArrIndex = taskData.findIndex(
+    (item) => item.id === task.parentElement.id
+  );
+  taskData.splice(dataArrIndex, 1);
+  localStorage.setItem("data", JSON.stringify(taskData));
+};
+
+const editTask = (task) => {
+  form.style.display = "block";
+  let dataArrIndex = taskData.findIndex(
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js:87]:
```suggestion
  const dataArrIndex = taskData.findIndex(
```

This doesn't appear to be reassigned.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/index.html [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        Add New Task
+      </button>
+
+      <!-- Modal -->
+      <form
+        class="modal"
+        id="modal"
+        tabindex="-1"
+        aria-labelledby="TODOModalLabel"
+        aria-hidden="true"
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/index.html:26]: We actually don't need to apply the `aria-hidden` property when the `display` is set to `none`.

<https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden>

Also, by applying this and not removing it in the onClick handler, the modal would never be visible to screen readers.

Comment (jdwilkin4) [reply javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/index.html:26]: ok, cool. That makes sense. 
I will remove it 👍

**Comment by naomi-lgbt** [comment]
I wonder if it would be worth exploring the [dialog](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) element in this project?

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/index.html [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        <form method="dialog">
+          <h2>Are you sure you want exit? All unsaved data will be lost</h2>
+          <div class="modal-btn-container">
+            <button class="btn" value="cancel" formmethod="dialog">
+              Cancel
+            </button>
+            <button class="btn" id="confirm-btn" value="default">Exit</button>
+          </div>
+        </form>
+      </dialog>
```
Comment (jdwilkin4) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/index.html:100]: @naomi-lgbt 

We are using the dialog element here

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+const openModalBtn = document.getElementById("open-modal-btn");
+const closeModalBtn = document.getElementById("close-modal-btn");
+const tasksContainer = document.getElementById("tasks-container");
+const textInput = document.getElementById("text-input");
+const dateInput = document.getElementById("date-input");
+const descriptionInput = document.getElementById("textarea");
+let taskData = [];
+
+openModalBtn.addEventListener("click", () => (form.style.display = "block"));
+closeModalBtn.addEventListener("click", () => confirmModal.showModal());
```
Comment (jdwilkin4) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js:13]: @naomi-lgbt 

The only time I was planning on mentioning the dialog element was here when we use the `showModal()` method.

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+/**
+ * We can have a debugging step where campers have to debug the error message of taskData.map is not a function
+ * We can have them initially log this to the console
+ * console.log(typeof localStorage.getItem("data"));
+ * They will see that it is a string which is why the error is there
+ * So then we can have them refactor it to use the JSON.parse method
+ *
+ */
+
+taskData = JSON.parse(localStorage.getItem("data")) || [];
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js:113]:
```suggestion
```

File: javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
+const form = document.getElementById("modal");
+const confirmModal = document.getElementById("confirm-modal");
+const confirmBtn = document.getElementById("confirm-btn");
+const openModalBtn = document.getElementById("open-modal-btn");
+const closeModalBtn = document.getElementById("close-modal-btn");
+const tasksContainer = document.getElementById("tasks-container");
+const textInput = document.getElementById("text-input");
+const dateInput = document.getElementById("date-input");
+const descriptionInput = document.getElementById("textarea");
+let taskData = [];
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-debugging-by-building-a-TODO-app/script.js:10]:
```suggestion
/**
 * We can have a debugging step where campers have to debug the error message of taskData.map is not a function
 * We can have them initially log this to the console
 * console.log(typeof localStorage.getItem("data"));
 * They will see that it is a string which is why the error is there
 * So then we can have them refactor it to use the JSON.parse method
 *
 */

const taskData = JSON.parse(localStorage.getItem("data")) || [];
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Last little nitpick and then this is good to go.

`taskData` was only reassigned globally, makes sense to initialise it (and use const) rather than reassign.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**